crates/decoder/src/decoder.rs

```diff
@@ -173,25 +168,34 @@ pub fn read_blocks_from_reader<R: Read>(
         .collect()
 }
 
-fn block_is_verified(block: &Block) -> bool {
-    if block.number != 0 {
-        if !block.receipt_root_is_verified() {
-            error!(
-                "Receipt root verification failed for block {}",
-                block.number
-            );
-            return false;
+fn block_is_verified(block: &AnyBlock) -> (bool, u64) {
+    match block {
+        AnyBlock::Evm(eth_block) => {
+            let block_number = eth_block.number;
+            if block_number != 0 {
+                if !eth_block.receipt_root_is_verified() {
+                    error!(
+                        "Receipt root verification failed for block {}",
+                        block_number
+                    );
+                    return (false, block_number);
+                }
+                if !eth_block.transaction_root_is_verified() {
+                    error!(
+                        "Transaction root verification failed for block {}",
+                        block_number
+                    );
+                    return (false, block_number);
+                }
+            }
+            (true, block_number)
         }
-
-        if !block.transaction_root_is_verified() {
-            error!(
-                "Transaction root verification failed for block {}",
-                block.number
-            );
-            return false;
+        // Logic is not yet implemented for verifying Solana Blocks
```

Comment:
I was a bit unclear on what was happening here. There are also `unimplemented!()` and `todo!()` macros you can use but not sure if that helps. Would be helpful to be clear on what this code is doing while waiting for more support.